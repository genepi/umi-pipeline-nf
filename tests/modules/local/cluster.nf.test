nextflow_process {

    name "Test Process CLUSTER"
    script "modules/local/umi_processing/cluster.nf"
    process "CLUSTER"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = Channel.of(
                    ["barcode02", "LPA2645", "$baseDir/tests/input/cluster/barcode02/detected_umis.fastq"],
                    ["barcode03", "LPA2645", "$baseDir/tests/input/cluster/barcode03/detected_umis.fastq"],
                    ["barcode04", "LPA2645", "$baseDir/tests/input/cluster/barcode04/detected_umis.fastq"],
                    ["barcode05", "LPA2645", "$baseDir/tests/input/cluster/barcode05/detected_umis.fastq"],
                )
                input[1] ="raw"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
