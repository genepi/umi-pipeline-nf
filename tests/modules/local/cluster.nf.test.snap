{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "barcode02",
                        "LPA2645",
                        "consensus.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "barcode03",
                        "LPA2645",
                        "consensus.fasta:md5,f5a74b69f8b7abf39df08abe2e2dcff3"
                    ],
                    [
                        "barcode04",
                        "LPA2645",
                        "consensus.fasta:md5,fe9d4a9c2dd4f828223e678244e92daa"
                    ],
                    [
                        "barcode05",
                        "LPA2645",
                        "consensus.fasta:md5,3a12b22ef0a847e788f5f5b06b3cdf2e"
                    ]
                ],
                "1": [
                    [
                        "barcode03",
                        "LPA2645",
                        [
                            "cluster0:md5,42f6543f76bc455c6fdf672506bec901",
                            "cluster1:md5,2ee339cd08b7e5427f95be618fc07224",
                            "cluster2:md5,3b5bf0296405e9949cc2004c951f1f6f",
                            "cluster3:md5,a49290bad5488423afac6202476f42d7",
                            "cluster4:md5,3a3e8d0766c1fc0e93f62144e7841acb",
                            "cluster5:md5,1c4f55a9cdde39329aba7f25c6455b34"
                        ]
                    ],
                    [
                        "barcode04",
                        "LPA2645",
                        [
                            "cluster0:md5,43434b02020cd2e41e33d92a10835f97",
                            "cluster1:md5,ddd4dd5d0918df43c48776d7082e9734",
                            "cluster2:md5,4bbd17234dd1ca4fd08baa11432dc8d8",
                            "cluster3:md5,ff0974f819897361226e1fc67aeb0656",
                            "cluster4:md5,f27185a6dcb557ed5cf56f3538941404",
                            "cluster5:md5,d042b1457bc3a906d6d6b24a72e6e980"
                        ]
                    ],
                    [
                        "barcode05",
                        "LPA2645",
                        [
                            "cluster0:md5,42f6543f76bc455c6fdf672506bec901",
                            "cluster1:md5,43434b02020cd2e41e33d92a10835f97",
                            "cluster10:md5,f27185a6dcb557ed5cf56f3538941404",
                            "cluster11:md5,d042b1457bc3a906d6d6b24a72e6e980",
                            "cluster2:md5,ddd4dd5d0918df43c48776d7082e9734",
                            "cluster3:md5,2ee339cd08b7e5427f95be618fc07224",
                            "cluster4:md5,3b5bf0296405e9949cc2004c951f1f6f",
                            "cluster5:md5,4bbd17234dd1ca4fd08baa11432dc8d8",
                            "cluster6:md5,ff0974f819897361226e1fc67aeb0656",
                            "cluster7:md5,a49290bad5488423afac6202476f42d7",
                            "cluster8:md5,3a3e8d0766c1fc0e93f62144e7841acb",
                            "cluster9:md5,1c4f55a9cdde39329aba7f25c6455b34"
                        ]
                    ]
                ],
                "cluster_fastas": [
                    [
                        "barcode03",
                        "LPA2645",
                        [
                            "cluster0:md5,42f6543f76bc455c6fdf672506bec901",
                            "cluster1:md5,2ee339cd08b7e5427f95be618fc07224",
                            "cluster2:md5,3b5bf0296405e9949cc2004c951f1f6f",
                            "cluster3:md5,a49290bad5488423afac6202476f42d7",
                            "cluster4:md5,3a3e8d0766c1fc0e93f62144e7841acb",
                            "cluster5:md5,1c4f55a9cdde39329aba7f25c6455b34"
                        ]
                    ],
                    [
                        "barcode04",
                        "LPA2645",
                        [
                            "cluster0:md5,43434b02020cd2e41e33d92a10835f97",
                            "cluster1:md5,ddd4dd5d0918df43c48776d7082e9734",
                            "cluster2:md5,4bbd17234dd1ca4fd08baa11432dc8d8",
                            "cluster3:md5,ff0974f819897361226e1fc67aeb0656",
                            "cluster4:md5,f27185a6dcb557ed5cf56f3538941404",
                            "cluster5:md5,d042b1457bc3a906d6d6b24a72e6e980"
                        ]
                    ],
                    [
                        "barcode05",
                        "LPA2645",
                        [
                            "cluster0:md5,42f6543f76bc455c6fdf672506bec901",
                            "cluster1:md5,43434b02020cd2e41e33d92a10835f97",
                            "cluster10:md5,f27185a6dcb557ed5cf56f3538941404",
                            "cluster11:md5,d042b1457bc3a906d6d6b24a72e6e980",
                            "cluster2:md5,ddd4dd5d0918df43c48776d7082e9734",
                            "cluster3:md5,2ee339cd08b7e5427f95be618fc07224",
                            "cluster4:md5,3b5bf0296405e9949cc2004c951f1f6f",
                            "cluster5:md5,4bbd17234dd1ca4fd08baa11432dc8d8",
                            "cluster6:md5,ff0974f819897361226e1fc67aeb0656",
                            "cluster7:md5,a49290bad5488423afac6202476f42d7",
                            "cluster8:md5,3a3e8d0766c1fc0e93f62144e7841acb",
                            "cluster9:md5,1c4f55a9cdde39329aba7f25c6455b34"
                        ]
                    ]
                ],
                "consensus_fasta": [
                    [
                        "barcode02",
                        "LPA2645",
                        "consensus.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "barcode03",
                        "LPA2645",
                        "consensus.fasta:md5,f5a74b69f8b7abf39df08abe2e2dcff3"
                    ],
                    [
                        "barcode04",
                        "LPA2645",
                        "consensus.fasta:md5,fe9d4a9c2dd4f828223e678244e92daa"
                    ],
                    [
                        "barcode05",
                        "LPA2645",
                        "consensus.fasta:md5,3a12b22ef0a847e788f5f5b06b3cdf2e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-02-19T14:37:52.28709"
    }
}